Add checks customization feature documentation (#211)
SUSE/doc-unversioned: trento/adoc/checks_customization.adoc

=== Target-Specific Check Customization

A check is always executed on a target, which can be a host or a cluster. Users can customize check values specific to the target environment to ensure optimal system performance. Customizations are target-specific and do not affect other targets or the default check values in the YAML file.

=== Required Permissions

Only *admin users* or users with the `all:checks_customization` permission can customize checks. Users without this permission will not see the customization button in the Trento Web console. If a check was modified, the modified Pill is shown to all users.

=== Customizable Checks

By default, a check with expected values is customizable. This applies to all checks unless explicitly disabled in the check YAML file. It cannot be changed in the Trento Web console.

Supported value types include:

> - Strings
> 
> - Numbers
> 
> - Booleans

Reviewer: ```suggestion
- String

- Number

- Boolean
```

User input is validated to ensure that the input value matches the expected type before allowing to save the custom values in Wanda's database. If the input type is incorrect or mixed, the customization fails, triggering a toast notification, that checks customization failed and a warning message in the modal itself.

> === Disable Check Customization

Author: This part is very optional, and less user relevant. Feel free to skip it, as discussed

> By default, built-in checks are customizable, to disable customization for a specific check, the *customization_disabled: true* flag must be added to the corresponding check YAML file. This flag can be applied at the top level to disable customization entirely or at the value level to control which values remain customizable through the Web UI.

Reviewer: ```suggestion
By default, built-in checks are customizable. To disable customization for a specific check, add the *customization_disabled: true* flag to the corresponding check YAML file. This flag can be applied at the top level to disable customization entirely or at the value level to control which values remain customizable through the Web UI.
```

[[checks_customization]]

== Checks Customization

=== Overview of checks Customization

> Trento makes it possible to adjust expected check values to match target-specific environmental requirements. This can be done directly through the Trento Web console without modifying the original check YAML files or creating separate checks for different environments.

Reviewer 2: ```suggestion
Trento makes it possible to adjust expected check values to match target-specific requirements. This can be done directly through the Trento Web console without modifying the original check or impacting other targets.
```

A check is always executed on a target, which can be a host or a cluster. Users can customize check values specific to the target environment to ensure optimal system performance. Customizations are target-specific, and they do not affect other targets or the default check values in the YAML file.

=== Required permissions

Only admin users and users with the `all:checks_customization` permission can customize checks. The customization button in the Trento Web console is not shown for users without this permission. If a check has been modified, the modified Pill is shown for all users.

=== Customizable checks

By default, all checks with expected values are customizable, unless they are explicitly disabled in the check YAML file. This cannot be changed in the Trento Web console.

Supported value types include:

- String
- Number
- Boolean

User input is validated to ensure that the input value matches the expected type before allowing to save the custom values in Wanda's database. If the input type is incorrect or mixed, the customization fails, triggering a toast notification, that checks customization failed and a warning message in the modal itself.

> === Disable Check Customization

Author: Hey @abravosuse , do you want to have at least the note about that it is possible to disable check customization in a check? Without going into details ?
